lib/Conversion/TorchToLinalg/TorchToLinalg.cpp

```diff
@@ -1512,7 +1512,13 @@ static Value createLinalgPayloadCalculationForElementwiseOp(
       mul.emitError("unimplemented: non-floating point dtype");
       return nullptr;
     }
-    return b.create<arith::MulFOp>(loc, payloadArgs[0], payloadArgs[1]);
+    AtenMulTensorOp::Adaptor adaptor(operands);
+    Type dtype = converter->convertType(mul.getType())
+                     .cast<RankedTensorType>()
+                     .getElementType();
+    Value lhs = convertScalarToDtype(b, loc, payloadArgs[0], dtype);
+    Value rhs = convertScalarToDtype(b, loc, payloadArgs[1], dtype);
+    return b.create<arith::MulFOp>(loc, lhs, rhs);
```

Comment:
Can you refer to https://github.com/llvm/mlir-npcomp/blob/5475d02a0d03b2bda7ef9cc240773041d22a926b/lib/Conversion/TorchToLinalg/TorchToLinalg.cpp#L1485 to add support for integer type as well.

Reply:
should I leave div alone for now? It seems like for integer operands both inputs should be promoted to float32:

https://pytorch.org/docs/stable/generated/torch.div.html

Reply:
ha, I see. yea, we can deal with div later.